src/components/ui/dialog.tsx

```diff
@@ -47,28 +43,42 @@ const DialogOverlay = ({
     />
   )
 }
-const DialogContent = React.forwardRef<
-  React.ElementRef<typeof DialogPrimitive.Content>,
-  DialogContentCustomProps
->(({ className, children, showClose = true, ...props }) => (
-  <DialogPrimitive.Content
-    data-slot="dialog-content"
-    className={cn(
-      'bg-background data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 fixed top-[50%] left-[50%] z-50 grid w-full max-w-[calc(100%-2rem)] translate-x-[-50%] translate-y-[-50%] gap-4 rounded-lg border border-border p-6 shadow-lg duration-200 sm:max-w-lg',
-      className
-    )}
-    {...props}
-  >
-    {children}
-    {showClose && <DialogPrimitive.Close className="ring-offset-background focus:ring-ring data-[state=open]:bg-accent data-[state=open]:text-muted-foreground absolute top-4 right-4 rounded-xs opacity-70 transition-opacity hover:opacity-100 focus:ring-2 focus:ring-offset-2 focus:outline-hidden disabled:pointer-events-none [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4">
-      <XIcon />
-      <span className="sr-only">Close</span>
-    </DialogPrimitive.Close>
-    }
-  </DialogPrimitive.Content>
-))
 
 
+interface DialogContentProps
+  extends React.ComponentProps<typeof DialogPrimitive.Content> {
+  customOverlay?: React.ReactNode,
+  showClose?: boolean
+}
+
+const DialogContent = ({
+  className,
+  children,
+  customOverlay,
+  showClose = true,
+  ...props
+}: DialogContentProps) => {
+  return (
+    <>
+      {customOverlay ? customOverlay : <DialogOverlay />}
```

Comment:
I think the DialogPortal needs to be dealt with still. The dialog in `AddPanel` is buggy, maybe because it is appended to the panel wrapper. In general, all dialogs should be the highest visible item, this is the reason why we hide everything behind the DialogOverlay. Users are "forced" to clear it first before continuing. To ensure this we should by default append the dialogs to `.tido` and in special cases receive a custom container.